server/api/check-story-changes.ts

```diff
@@ -0,0 +1,141 @@
+import type { Request, Response } from 'express';
+import { validateAtlassianHeaders, handleApiError, type ErrorCommentContext } from './api-error-helpers.js';
+import { createAtlassianClientWithPAT, type AtlassianClient } from '../providers/atlassian/atlassian-api-client.js';
+import { createProviderFromHeaders } from '../llm-client/index.js';
+import { executeCheckStoryChanges as defaultExecuteCheckStoryChanges, type ExecuteCheckStoryChangesParams } from '../providers/combined/tools/check-story-changes/core-logic.js';
+import type { ToolDependencies } from '../providers/combined/tools/types.js';
+import { resolveCloudId } from '../providers/atlassian/atlassian-helpers.js';
+import { 
+  createProgressCommentManager,
+  type ProgressCommentManager 
+} from './progress-comment-manager.js';
+
+/**
+ * Dependencies that can be injected for testing
+ */
+export interface CheckStoryChangesHandlerDeps {
+  executeCheckStoryChanges?: (params: ExecuteCheckStoryChangesParams, deps: ToolDependencies) => Promise<any>;
+  createAtlassianClient?: typeof createAtlassianClientWithPAT;
+}
+
+/**
+ * POST /api/check-story-changes
+ * 
+ * Analyze divergences between a child story and its parent epic
+ * 
+ * Headers:
+ *   X-Atlassian-Token: ATATT...  (Atlassian PAT)
+ *   X-Atlassian-Email: user@example.com  (Email for Basic Auth)
+ *   X-Anthropic-Token: sk-...    (Anthropic API key)
+ * 
+ * Request body:
+ * {
+ *   "storyKey": "PROJ-124",
+ *   "siteName": "my-jira-site",  // optional
+ *   "cloudId": "uuid"            // optional
+ * }
+ * 
+ * Response:
+ * {
+ *   "success": true,
+ *   "analysis": "# Divergence Analysis\n\n...",
+ *   "metadata": {
+ *     "parentKey": "PROJ-100",
+ *     "childKey": "PROJ-124",
+ *     "tokensUsed": 500
+ *   }
+ * }
+ * 
+ * @param deps - Optional dependencies for testing (defaults to production implementations)
+ */
+export async function handleCheckStoryChanges(req: Request, res: Response, deps: CheckStoryChangesHandlerDeps = {}) {
+  // Use injected dependencies or defaults
+  const executeCheckStoryChangesFn = deps.executeCheckStoryChanges || defaultExecuteCheckStoryChanges;
+  const createAtlassianClientFn = deps.createAtlassianClient || createAtlassianClientWithPAT;
+  
+  let commentContext: ErrorCommentContext | null = null;
+  let progressManager: ProgressCommentManager | null = null;
+  
+  try {
+    const { cloudId, siteName } = req.body;
+
+    // Validate storyKey
+    const storyKey = req.body.storyKey;
+    if (!storyKey) {
+      res.status(400).json({ success: false, error: 'Missing required field: storyKey' });
+      return;
+    }
+
+    const tokens = validateAtlassianHeaders(req.headers, res);
```

Comment:
How are other api's validating headers? I'm surprised you need to create a new helper